plugins/rest/rest_test.go

```diff
@@ -1912,11 +1917,15 @@ func TestDebugLoggingRequestMaskAuthorizationHeader(t *testing.T) {
 	for _, entry := range logger.Entries() {
```

Comment:
suggestion: if you fancy, I think we can make this test a little clearer. At the moment, if this test fails we get `Expected log entry from request` which isn't very helpful.

`logger.Entries()` will be two entries, the request and the response in that order. I think it'd make sense to only test the contents of the request entry since the masking only applies to that step.

i.e.

```go
	entries := logger.Entries()
	if len(entries) != 2 {
		t.Fatalf("Expected 2 log entries, got %d", len(entries))
	}

	requestEntry := entries[0]
```

Then we can make three checks on `requestEntry`: that both the `Authorization` header and the `X-AMZ-SECURITY-TOKEN` headers are set to `REDACTED` and that the `Remains-Unmasked` is set to the plaintext value.

We can always do this another time if you'd rather.

Reply:
Agreed, I'll make this change.